### ISSUE TITLE
Implement simple custom Layer

### ISSUE THREAD

**Issue by ralfriegel5** [keras-team/keras#9878]
I want to implement a simple custom layer like this:

![graphlayer](https://user-images.githubusercontent.com/19731019/38466105-58cafe76-3b24-11e8-9c24-893f16aa002e.jpg)

The X Layer has many neurons (> 3000), so there is no chance to simply create some 3-neuron-dense-layers and then connect them using the functional model.
Do you have any idea how to do this in a smart way, so it will run on the GPU?

Thanks!

**Comment by nuric** [comment]
That looks like a [Conv1D](https://keras.io/layers/convolutional/#conv1d) or [LocallyConnected1D](https://keras.io/layers/local/) depending on whether you want the weights shared or not with a stride of 3. I'm not sure why you would need a custom layer. This is an implementation related question; so you'll still get a better, cleaner answer on [StackOverflow](http://stackoverflow.com/questions/tagged/keras).

**Comment by rex-yue-wu** [comment]
It actually is straight-forward to implement this layer. All you need is to follow the instruction of creating a custom layer. 

```python
from keras.layers import Layer, Dense
from keras import backend as K

class DenseK( Layer ) :
    def __init__( self, K=3, **kwargs ) :
        self.K = K
        super( DenseK, self ).__init__( **kwargs )
    def build( self, input_shape ) :
        feat_dims = input_shape[-1]
        assert feat_dims % self.K == 0
        self.kernel = self.add_weight(shape=( 1, feat_dims ),
                                      initializer='glorot_uniform',
                                      name='kernel' )
        self.bias = self.add_weight(shape=( 1, feat_dims//self.K ),
                                      initializer='zeros',
                                      name='bias' )
        self.xk_shape = (-1, feat_dims//self.K, self.K )
    def call( self, x ) :
        # 1. element-wise product between x and kernel
        xk = x * self.kernel
        # 2. reshape xk, xk.shape = (batch_size, input_feat_dim//3, 3 )
        xk = K.reshape( xk, self.xk_shape ) 
        # 3. compute y for every K elements in xk
        y = K.sum( xk, axis=-1, keepdims=False ) + self.bias
        return y 
    def compute_output_shape( self, input_shape ) :
        batch_size, feat_dims = input_shape
        return ( batch_size, feat_dims//self.K )
```
Then you can verify this layer with some testing code like below.
```python
from keras.layers import Input
from keras.models import Model
x = Input(shape=(3000,))
y = DenseK(K=3, name='3-neuron-dense')(x)
model = Model(inputs=x, outputs=y)
print model.summary()

import numpy as np 
a = np.random.randn(2,3000)
b = np.random.randn(2,1000)

model.compile('sgd', loss='mse')
model.fit(a,b)
```

Of course, this is a very basic implementation and you can add more supports, e.g. activation function. 

I think the most important thing in this implementation is how to translate your goal into a series of tensor operations. There are many ways to achieve your goal, but I think the one I picked is probably the fastest and simplest.

**Comment by gauravkr0071** [comment]
import keras
from keras.layers import Input, Dense
from keras.models import Model
import numpy as np
import matplotlib.pyplot as plt
XOR_data=np.array([[0.1,0.1,1],   #--->1st set
                   [0.1,0.9,1],   #--->2nd set            ## used 0.1 instead of 0 for better learning
                   [0.9,0.1,1],   #--->3rd set            ## used 0.9 instead of 1 for better ---
                   [0.9,0.9,1]    #--->4th set
                  ]) 
XOR_data=XOR_data.reshape(4,1,3)

y_true=np.array([[0.1],      #--->y_true for 1st set
                 [0.9],      #--->y_true for 2nd set
                 [0.9],      #--->y_true for 3rd set
                 [0.1]       #--->y_true for 4th set
                ])  
y_true=y_true.reshape(4,1,1)

img_shape=(1,3)
input_vec = keras.Input(shape=img_shape)



output_1 = Dense(4, activation='sigmoid')(input_vec) 
output_2 = Dense(2, activation='sigmoid')(output_1 )# replace this one with custom layer 
y_pred = Dense(1, activation='sigmoid')(output_2)


model = Model(input_vec,y_pred)
model.summary()
model.compile(loss='mse',optimizer='adam',metrics=['accuracy'])

can any1 help me making a dense layer but with custom layer , just for practise so i can know how to make custom layer in keras

**Comment by gauravkr0071** [comment]
![git11](https://user-images.githubusercontent.com/51910127/75352671-237c1e00-58d0-11ea-932d-db4bf313b70a.PNG)
hey frnds, i am just a beginner , i want to make a dense layer in keras but with custom layer so i can work on more complex thing ? 
how to make it  ? just give me necessary code which i can easily put in between
